lib/queue.js

```diff
@@ -41,21 +30,79 @@ function compare (a, b) {
     return matches;
 }
 
+/**
+ * Creates a new queue
+ * @class
+ */
 var FetchQueue = function() {
-    this.oldestUnfetchedIndex = 0;
-    this.completeCache = 0;
-    this.scanIndex = {};
+    Array.call(this);
+
+    /**
+     * Speeds up {@link FetchQueue.oldestUnfetchedItem} by storing the index at
+     * which the latest oldest unfetched queue item was found.
+     * @name FetchQueue._oldestUnfetchedIndex
+     * @private
+     * @type {Number}
+     */
+    Object.defineProperty(this, "_oldestUnfetchedIndex", {
+        enumerable: false,
+        writable: true,
+        value: 0
+    });
+
+    /**
+     * Serves as a cache for what URL's have been fetched. Keys are URL's,
+     * values are booleans.
+     * @name FetchQueue._scanIndex
+     * @private
+     * @type {Object}
+     */
+    Object.defineProperty(this, "_scanIndex", {
+        enumerable: false,
+        writable: true,
+        value: {}
+    });
+
+    /**
+     * Controls what properties can be operated on with the
+     * {@link FetchQueue#min}, {@link FetchQueue#avg} and {@link FetchQueue#max}
+     * methods.
+     * @name FetchQueue._allowedStatistics
+     * @type {Array}
+     */
+    Object.defineProperty(this, "_allowedStatistics", {
+        enumerable: false,
+        writable: true,
+        value: [
+            "actualDataSize",
+            "contentLength",
+            "downloadTime",
+            "requestLatency",
+            "requestTime"
+        ]
+    });
 };
 
-module.exports = FetchQueue;
+FetchQueue.prototype = Array.prototype;
 
-FetchQueue.prototype = [];
```

Comment:
Previously, if for some reasons multiple queue instances would have been created, they would've inherited from the same array. Now they inherit from `Array.prototype` instead